### ISSUE TITLE
Types appearing both in module and in exports

### ISSUE THREAD

**Issue by arlyon** [bytecodealliance/wit-bindgen#903]
Consider this wit

```wit
package litehouse:plugin;

interface plugin {
    variant update {
      time(u64),

      temperature(float64),
      wind-speed(float64),

      current(float64),
      voltage(u16),
      power(u16),
      on-off(bool),
    }
}

world plugin-host {
  use plugin.{update};
  import send-update: func(nickname: string, update: update);

  export plugin;
}

```

I want to simultaneously expose the interface to the host as well as require the host to expose a function for the guest to use. The generated code however produces two types, `bindings::litehouse::plugin::plugin::Update` and `bindings::exports::litehouse::plugin::plugin::Update`, one presumably stems from the `use` and one from the `export`. These types are (to my eyes) identical. Any `Update` coming from the guest has the first, and any `Update` going into the guest has the latter. I would assume we would be able to, in the case of an interface being exported, defer all references of it to its exported types. Is there a technical reason for not doing this? If it is some technical issue, then a From / Into each way would be handy.

**Comment by alexcrichton** [comment]
Nice! I think that the dtor issue may have been fixed in https://github.com/bytecodealliance/wit-bindgen/pull/946, but mind confirming?

**Comment by ydnar** [comment]
I'll try this out, thanks!

Does this also apply to resource.new/resource.rep?

**Comment by alexcrichton** [comment]
I don't think so, but I'd have to dig in to be sure. I say this though because resource new/rep only work on exported resources, so I don't think the `[export]` is needed to disambiguate there (as it's always exports)

**Comment by ydnar** [comment]
Is there value in consistency for imports on exported types?

**Comment by alexcrichton** [comment]
I don't have much of a preference myself, but it seems reasonable to at least accept both perhaps
